src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs

```diff
@@ -1122,11 +1101,19 @@ internal void FlushContentToDisk()
                     {
                         try
                         {
+                            Encoding currentEncoding;
```

Comment:
Since we don't initialize the variable we can move the definition to line 1109

Reply:
Because of the using we can't, and I can't put the rest of the code inside of the using otherwise the StreamWriter will not have access to the file because it will be in use.